### ISSUE TITLE
FunctionMaxLength reports long test functions

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#4921]
## Current Behavior
This is flagged
```kotlin
@Test
fun testFallbackSendsNoLocationEvenIfUpdatedAfterTimeout() {
```

## Expected Behavior
Not flagged, and I shouldn't have to configure this as this will be an issue in all projects out there. or 80/20 if there are some places where long test names are discouraged.

## Context
Following up on https://github.com/detekt/detekt/issues/2745, how about by default ignoring on Test annotations?
```diff
  FunctionMaxLength:
    active: true
+   ignoreAnnotated: 'Test'
    maximumFunctionNameLength: 30
```

Note that this would be mostly library-agnostic, as most JVM testing libraries use the Test annotation. We could even argue that this is Kotlin's native stuff as the kotlin-test library also aliases the Test annotation from JUnit for example.

**Comment by TWiStErRob** [comment]
Note: alternatively maybe allow somehow to specify different length restriction for tests.

**Comment by cortinico** [comment]
I had a similar scenario for this rule in one of my project.
I think it's safe to exclude this rule for tests using the `exclude:` field as we do for other rules as well.

**Comment by TWiStErRob** [comment]
Since we have a more surgical way of suppressing now, why would we go for the blanket ignore?

**Comment by cortinico** [comment]
> Since we have a more surgical way of suppressing now, why would we go for the blanket ignore?

Because is more consistent with the rest of the rules we already support.

**Comment by TWiStErRob** [comment]
https://github.com/detekt/detekt/discussions/4924

**Comment by cortinico** [comment]
> To be honest I don't know why we have this rule in first place. It's true that it's really easy to implement and maintain but it provides nearly 0 value.

I think it still has value in the sense that it forces users to don't write method names that are too long and complicated (imho is a code smell in some form).

**Comment by gouri-panda** [comment]
Should I work on this issue?

**Comment by cortinico** [comment]
> I think it's safe to exclude this rule for tests using the `exclude:` field as we do for other rules as well.

@gouri-panda I would suggest we solve this in this way. Opinions?

**Comment by gouri-panda** [comment]
@cortinico  ok 👍

**Comment by TWiStErRob** [comment]
Good enough, @cortinico 👍

**Comment by chao2zhang** [comment]
Closing as the current configs provide a decent way to address this.

**Comment by TWiStErRob** [comment]
They do, it's just the question of DX-friendly defaults.
